indexdata.go

```diff
@@ -52,6 +55,8 @@ type indexData struct {
 	fileNameIndex   []uint32
 	fileNameNgrams  map[ngram][]uint32
 
+	fileEndSymbol []uint32
```

Comment:
why do you need fileEndSymbol? Can't we just rely on the same indexes as our document sections in the matching code?